### PR TITLE
Feature/junit reporter

### PR DESCRIPTION
**Pull request by awinder**
# Description

This adds a new flag to bru run, the `--format` flag. It defaults to `"json"`, which is the current output file format, while also implementing a `"junit"` option that will produce JUnit-formatted XML.

There are some opinionated implementation choices which I don't have strong opinions about:

1. Using the bru filename (minus `.bru`) for test suite names in the junit markup
2. Using a structure where the filename becomes a JUnit test suite, and then assertions + tests become test cases inside the suite
3. JUnit tests (often? must?) supply a `time` attribute, and I had access to timing details on the .bru file / test suite level, but not the test case level. For implementation simplicity what I did was I used timing details from the .bru file to set the `time` attribute on the test suite, and then I divided it equally for the child testcases (which are the assertions / individual tests). This seemed accurate enough while hopefully not breaking some tooling that expects the `time` attr to be present

Also, I don't think I messed anything up applying a new npm module for xml writing (and chose a module which was already being required outside of the cli package), but the package-lock.json file updated widely so probably best to double-check there. I also removed a `@jest/globals` require in the test file because I couldn't get that to work locally, I can add that back, but I'm pretty sure it's unnecessary? And of course happy to address any feedback.

MOST IMPORTANTLY: THANK YOU so much for your work on Bruno, this project is awesome. I have now been burned twice on tools in this family (once with Swagger, and now again with Insomnia) adopting cloud policies that will never be acceptable as a business risk. I want to figure out how to champion sponsorship for this project but until then, consider this a small token of appreciation 😄 

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

### CODE REVIEW

File: packages/bruno-cli/src/commands/run.js [hunk 1]
```diff
@@ -165,6 +167,73 @@ const getCollectionRoot = (dir) => {
...
+      '@timestamp': new Date().toISOString().split('Z')[0],
+      '@hostname': os.hostname(),
+      '@time': result.runtime.toFixed(3),
+      testcase: []
+    };
+
+    result.assertionResults &&
+      result.assertionResults.forEach((assertion) => {
+        const testcase = {
+          '@name': `${assertion.lhsExpr} ${assertion.rhsExpr}`,
```
Comment (ngoetje) [inline packages/bruno-cli/src/commands/run.js]: How about adding the operator to the name, I think that is important to see the actual assertion that is tested here.

Comment (awinder) [reply packages/bruno-cli/src/commands/run.js]: In testing this it seems like the operator is already included in the `rhsExpr` field? When I ran this against some of my existing test suites I got results like `res.body.array_field.length gt 0` or `res.body[0].field_name isString`. Are there some conditions where the operator isn't included in the rhsExpr?

**Comment by helloanoop** [comment]
Thanks for the kind words @awinder 💛 

PR looks good to me. I would recommend once change before we merge this
Let's move the junit reporter logic to a separate file.
I am thinking of a structure like
```
src
  |--commands
  |--runner
  |--reporters
    |--junit.js
  |--utils
```

There are some conflicts since I merged some PRs. Please sync your fork with the main.

**Comment by awinder** [comment]
Awesome, good idea. I refactored this & cleared up the merge conflicts but let me know if there's anything else!

**Comment by helloanoop** [comment]
Merged!

Thank you @awinder !

**Comment by helloanoop** [comment]
@awinder I have released Bru CLI version `1.3.0`

**Comment by AkkafaOguz** [comment]
Hi @awinder, I hope you're well. First of all, thanks for your great effort!
I've tried to use 'bru run <folderName> --env <environmentName> --output result.xml --format junit' command. However, I'm getting "Unknown argument: format" message. What might be the issue? Am I doing something wrong? Your works will help a lot, thanks!

**Comment by AkkafaOguz** [comment]
Hi @awinder, I hope you're well. First of all thanks for your great effort!
I want to be sure that the command for format output feature is "bru run <folderName> --env <envName> --output response.xml --format junit".  Because when I use this command, I got "Unknown argument: format" message. Thanks for your interest and time.

**Comment by awinder** [comment]
> Hi @awinder, I hope you're well. First of all thanks for your great effort! I want to be sure that the command for format output feature is "bru run `<folderName>` --env `<envName>` --output response.xml --format junit". Because when I use this command, I got "Unknown argument: format" message. Thanks for your interest and time.

Hi @AkkafaOguz, you might need to upgrade the bru CLI to the latest version. Check `bru --version` and make sure it's > 1.3.0. Then you can run `npm update -g @usebruno/cli` if you're out-of-date to pick up this feature. Your command works locally after I removed & reinstalled so I'm pretty sure this is a version issue.

**Pull request merged**